### PR TITLE
Basic D3DContext prototype application

### PR DESCRIPTION
**Pull request by AlexB8675**
Before I go too forward, we should decide on design matters such as:
- MakeGraphicsPipeline in D3DContext or Initialize in GraphicsPipeline
- Is creating functions that wrap around D3D11 calls necessary or should we just use D3DContext directly
- Should a custom class for a graphics pipeline exist in the first place
- Should we make D3DContext a unique_ptr

### CODE REVIEW

**Comment by NicusorN5** [comment]
I vote for
-> directly calling DX methods/functions
-> no custom class
-> I vote for the unique_ptr, yes

**Comment by AlexB8675** [comment]
Please check now, I think abstracting away things that may get repetitive such as vertex buffer creation and pipeline creation is good, while in rendering, I call into D3D11 directly, you can check now for the hello triangle example.

File: src/Cpp/1-getting-started/11-HelloTriangle/HelloTriangle.cpp [hunk 1]
```diff
@@ -51,7 +54,10 @@ bool HelloTriangleApplication::Initialize()
         0u, 1u, 2u
     };
     _triangle = std::make_unique<StaticMesh>();
-    if (!_dxContext->MakeStaticMesh(std::move(vertices), std::move(indices), _triangle.get()))
+    if (!_dxContext->MakeStaticMesh(
```
Comment (Themperror) [inline src/Cpp/1-getting-started/11-HelloTriangle/HelloTriangle.cpp:57]: not a fan of moving into MakeStaticMesh and then in there moving it back into _triangle, maybe change the API and do:
```cpp
_triangle.vertices = {
         0.0f,  0.5f, 0.0f,
         0.5f, -0.5f, 0.0f,
        -0.5f, -0.5f, 0.0f,
    };
    _triangle->indices =  {
        0u, 1u, 2u
    };
   _triangle->MakeStaticMesh() ;
   ```
   instead? 
   
   Or just perhaps accept the potential copy (or rely on copy-elision) and give normal const&'s instead

**Comment by NicusorN5** [comment]
> Please check now, I think abstracting away things that may get repetitive such as vertex buffer creation and pipeline creation is good, while in rendering, I call into D3D11 directly, you can check now for the hello triangle example.

Yes, but remember that the purpose is learning DX11. In a real case you'd wrap DX for your own sake.

**Comment by AlexB8675** [comment]
> > Please check now, I think abstracting away things that may get repetitive such as vertex buffer creation and pipeline creation is good, while in rendering, I call into D3D11 directly, you can check now for the hello triangle example.
> 
> Yes, but remember that the purpose is learning DX11. In a real case indeed you'd wrap DX for your own sake.

Why not just show them in the tutorial as a function, instead of writing the same code over and over again?

File: src/Cpp/1-getting-started/11-HelloTriangle/HelloTriangle.cpp (hunk 1 above)
Comment (AlexB8675) [reply src/Cpp/1-getting-started/11-HelloTriangle/HelloTriangle.cpp:57]: I agree, I will push a fix right now

**Comment by AlexB8675** [comment]
closing, needs further review

**Pull request closed without merging**